### DISCUSSION TITLE
Number of Agents

### DISCUSSION THREAD

**Discussion by leenr01** [gama-platform/gama#74] [Q&A] [answered]
What is the maximum number of agents that GAMA can deal with?
I want to study a complex system with at least 10,000 agents in GAMA. However, whenever the number of agents is more than 1500 the code gets impractically slow.

**Comment by lesquoyb** [comment, marked as answer]
If there's any hard limit, it for sure is way higher than that, I was able to instantiate simple models with 1 000 000 agents without any problem. But then you'll encounter your computer's limits.
So I guess your problem here is just that your model is slow to compute. There are some tips [here](https://gama-platform.org/wiki/next/OptimizingModels) about model optimization
